libbeat/common/kubernetes/k8skeystore/kubernetes_keystore.go

```diff
@@ -101,6 +101,9 @@ func NewKubernetesSecretsKeystore(keystoreNamespace string, ks8client k8s.Interf
 func (k *KubernetesSecretsKeystore) Retrieve(key string) (*keystore.SecureString, error) {
 	// key = "kubernetes.somenamespace.somesecret.value"
 	tokens := strings.Split(key, ".")
+	if len(tokens) > 0 && tokens[0] != "kubernetes" {
+		return nil, keystore.ErrKeyDoesntExists
+	}
```

Comment:
Does this fix need a changelog entry? Should it be done in a different PR?

Reply:
This one is mostly to "mute" some DEBUG messages that are shown in the console when configs are being evaluated at https://github.com/elastic/beats/blob/e523380e7b51bc482f78428c7c54de0de25dc2e4/libbeat/autodiscover/template/config.go#L135.

With this addition we make sure that we don't log DEBUG msgs for keys that come from unrelated configs (other than `kubernetes.*`)